src/frontend/js/widgets/CourseProductItem/components/PurchaseButton/index.spec.tsx

```diff
@@ -74,14 +74,37 @@ describe('PurchaseButton', () => {
     expect(screen.queryByTestId('SaleTunnel__modal')).toBeNull();
 
     // Then user can enter into the sale tunnel and follow its 3 steps
-    await act(async () => {
+    act(() => {
       fireEvent.click(button);
     });
 
     // - SaleTunnel should have been opened
     screen.getByTestId('SaleTunnel__modal');
   });
 
+  it('renders a disabled CTA if the product have no remaining orders', async () => {
```

Comment:
Can you also write a test to ensure that all is working as expected when `remaining_order_count` is `null`.